packages/app/src/cli/models/extensions/specifications/payments_app_extension_schemas/base_payments_app_extension_schema.ts

```diff
@@ -0,0 +1,45 @@
+import {BaseSchema} from '../../schemas.js'
+import {zod} from '@shopify/cli-kit/node/schema'
+
+const MAX_LABEL_SIZE = 50
+
+export const BasePaymentsAppExtensionSchema = BaseSchema.extend({
```

Comment:
I like the approach of using composition to build each schema using logical groupings like `DeferredPaymentsSchema` and `ConfirmationSchema`. However, I am not so sure if lumping a group of fields under `BasePaymentsAppExtensionSchema` in aligned with that idea. I think it's fine to have some duplicate fields in each extension schema if there isnt a logical grouping associated with those. What do you think?

Reply:
> I think it's fine to have some duplicate fields in each extension schema if there isnt a logical grouping associated with those. Nonetheless, it isnt a sticking point. What do you think?

That breaks the idea of lifting common fields into a single schema. Having schemas being defined as `BaseSchema.merge(PaymentsAppUrlExtensionsSchema).merge(MerchantLabelSchema).merge(InputSchema).merge(TestModeSchema)` seems insane to me. That's a lot of boilerplate code for just strictly sticking to composition.

I think we get the best of both worlds by using a common schema to all payment apps (`BasePaymentsAppExtensionSchema`), and then, to prevent code duplication, using schema "plugins" to extend the available fields. 

It's what we already do for the common `BaseSchema` schema, common to all extensions

Reply:
I think the extracting out the common properties has come at a slight cost in terms of code readability but I understand the approach. approving the PR !